src/lib/acf/ut/CMakeLists.txt

```diff
@@ -10,7 +10,7 @@ add_executable(${test_app} ${acf_ut_srcs})
 set_property(TARGET ${test_app} PROPERTY FOLDER "app/tests")
 target_link_libraries(${test_app} PUBLIC acf GTest::gtest ${OpenCV_LIBS})
 
-if ((${ACF_DO_GPU_TESTING}) AND ACF_BUILD_OGLES_GPGPU)
+if (ACF_BUILD_OGLES_GPGPU and aglet_FOUND)
```

Comment:
May be `AND TARGET aglet::aglet`?

Reply:
👍

Reply:
In this approach it uses CMake logic at a higher level to determine whether or not aglet will be available and then adds the package that will be shared by different internal targets.  The downstream test (and soon the application) can then test whether or not aglet is available in order to enable the OpenGL stuff.  The other approach would be to move the CMake logic to a lower level and have the test and the application independently add aglet (without overhead).   I guess the later approach would be similar to "include what you use".   I'm not sure what is the *right* approach.  As the project structure gets more complicated (more targets, etc) I can see merits to "include what you use".  I'll update with the `TARGET aglet::aglet` check for now.

Reply:
I think it's better to use:
```cmake
if(CONDITION_A AND CONDITION_B)
  hunter_add_package(aglet)
  find_package(aglet CONFIG REQUIRED)
  target_link_libraries(foo PUBLIC aglet::aglet)
  target_compile_definitions(foo PUBLIC FOO_WITH_AGLET)
endif()
```

because if you spread/separate logic it will not scale, e.g.
```cmake
if(TARGET_X_CONDITION)
  hunter_add_package(aglet)
  find_package(aglet CONFIG REQUIRED)
endif()

if(TARGET_Y_CONDITION)
  hunter_add_package(aglet)
  find_package(aglet CONFIG REQUIRED)
endif()

if(TARGET aglet::aglet)
  target_link_libraries(x PUBLIC aglet::aglet) # wrong: will be true if TARGET_X_CONDITION=OFF and TARGET_Y_CONDITION=ON
  target_link_libraries(y PUBLIC aglet::aglet) # wrong: will be true if TARGET_Y_CONDITION=OFF and TARGET_X_CONDITION=ON
endif()
```